google-cloud-bigquery/lib/google/cloud/bigquery/schema.rb

```diff
@@ -276,141 +244,86 @@ def date name, description: nil, mode: :nullable
         #   end
         #
         def record name, description: nil, mode: nil
+          # TODO: do we need to fail if no block was given?
```

Comment:
Based on the definition set here: https://cloud.google.com/bigquery/data-types we might want to even ensure there is at least on field in the record.